pandas/io/excel/_base.py

```diff
@@ -1014,16 +1015,22 @@ def close(self):
         return content
 
 
-XLS_SIGNATURE = b"\xD0\xCF\x11\xE0\xA1\xB1\x1A\xE1"
-ZIP_SIGNATURE = b"PK\x03\x04"
-PEEK_SIZE = max(len(XLS_SIGNATURE), len(ZIP_SIGNATURE))
+SIGNATURES = {
+    "biff2": b"\x09\x00\x04\x00",
+    "biff3": b"\x09\x02\x06\x00",
+    "biff4": b"\x09\x04\x06\x00",
+    "biff5": b"\xD0\xCF\x11\xE0\xA1\xB1\x1A\xE1",
+    "biff8": b"\xD0\xCF\x11\xE0\xA1\xB1\x1A\xE1",
+    "zip": b"PK\x03\x04",
+}
+PEEK_SIZE = max(map(len, SIGNATURES.values()))
 
 
 @doc(storage_options=_shared_docs["storage_options"])
 def inspect_excel_format(
     content_or_path: FilePathOrBuffer,
     storage_options: StorageOptions = None,
-) -> str:
+) -> Union[str, None]:
```

Comment:
Please str | None, this is the new mechanism to type this

Reply:
Oh, thanks for the heads up: I looked around in the typing documentation and didn't see this feature. Will adjust now.

Reply:
see https://docs.python.org/3.10/whatsnew/3.10.html#pep-604-new-type-union-operator if you are interested